### PR TITLE
struct generated for component should have public visibility

### PR DESCRIPTION
**Pull request by zengsai**

### CODE REVIEW

File: codegen/src/component.rs [hunk 1]
```diff
@@ -194,7 +194,7 @@ impl ComponentMeta {
 
             Some(quote! {
                 #[derive(Clone)]
-                struct #ident(std::rc::Rc<std::cell::RefCell<ruukh::component::Status<#state_ty>>>);
+                pub struct #ident(std::rc::Rc<std::cell::RefCell<ruukh::component::Status<#state_ty>>>);
```
Comment (csharad) [inline codegen/src/component.rs]: It seems you are making the status struct public by default. We should rather inherit the visibility from the component struct.

File: codegen/src/component/state.rs [hunk 2]
```diff
@@ -51,7 +51,7 @@ impl StateMeta {
                 self.expand_fields_with(ComponentField::to_field_assignment_as_default);
 
             Some(quote! {
-                struct #ident {
+                pub struct #ident {
```
Comment (csharad) [inline codegen/src/component/state.rs]: Here too we should inherit the visibility of the component.

**Comment by csharad** [comment]
Thank you for your PR.
I have posted in reviews regarding the default visibility. So, fix them and if you have any questions, throw them at me.
Thanks again :smile: .

**Comment by zengsai** [comment]
Updated. Please review again.

**Comment by csharad** [comment]
Thanks again @zengsai.

**Pull request merged**